### ISSUE TITLE
Facing below issue during remidiation

### ISSUE THREAD

**Issue by Nikhleshk** [gettek/terraform-azurerm-policy-as-code#25]
╷    
│ Error: Invalid for_each argument
│
│   on ..\modules\set_assignment\main.tf line 36, in resource "azurerm_management_group_policy_remediation" "rem":
│   36:   for_each                = { for dr in local.definition_reference.mg : basename(dr.reference_id) => dr }
│     ├────────────────
│     │ local.definition_reference.mg will be known only after apply
│
│ The "for_each" value depends on resource attributes that cannot be determined until apply, so Terraform     
│ cannot predict how many instances will be created. To work around this, use the -target argument to first   
│ apply only the resources that the for_each depends on.

**Comment by gettek** [comment]
Please do not remove the Issue Template when creating a new issue

**Comment by Nikhleshk** [comment]
Can you help me with this?

**Comment by Nikhleshk** [comment]
Ok
